### PR TITLE
add fyi about mismatched config values

### PR DESCRIPTION
**Pull request by wkozyra95**
fyi page that could be linked in error here https://github.com/expo/eas-build/pull/228

### CODE REVIEW

**Review by szdziedzic — APPROVED** [review]
Thanks for adding this page!

File: eas-config-mismatch.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Evaluating app config on EAS Build workers.
+
+Part of the build process on EAS Build is evaluation of the content of **app.json/app.config.js** file. In some cases, if the project is misconfigured, resulting config object might have an incorrect value e.g. build might happen on the project A, but `extra.projectId` might refer to the project B.
```
Comment (brentvatne) [inline eas-config-mismatch.md:3]:
```suggestion
Part of the build process on EAS Build is evaluation of the content of **app.json/app.config.js** file. In some cases, if the project is misconfigured, the resulting config object may contain incorrect values e.g. the build is running under project A, but `extra.projectId` refers to project B.
```

File: eas-config-mismatch.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+
+### Building from EAS CLI
+
+When building from EAS CLI we are using both environment variables from your local environment and values from **eas.json**, but on the EAS Build worker only values from **eas.json** (and EAS Secrets) will be present. It's common to use dynamic app config (**app.config.js/app.config.ts**) to switch between different configuration based on environment variables. When building on EAS, users need to make sure that environment on the EAS Build will match what was set locally.
```
Comment (brentvatne) [inline eas-config-mismatch.md:8]:
```suggestion
When building from EAS CLI we are using both environment variables from your local environment and values from **eas.json**, but on the EAS Build worker only values from **eas.json** and EAS Secrets will be present. It's common to use dynamic app config (**app.config.js/app.config.ts**) to switch between different configuration based on environment variables. When building on EAS, users need to make sure that environment on the EAS Build will match what was set locally.
```

File: eas-config-mismatch.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+If you have **app.config.js**:
+
+```js
+let projectId;
+if (process.env.APP === 'app1') {
+    projectId = '23847707-bf15-424c-b1dc-7a57fc34a23d';
```
Comment (brentvatne) [inline eas-config-mismatch.md:15]:
```suggestion
  projectId = '23847707-bf15-424c-b1dc-7a57fc34a23d';
```

File: eas-config-mismatch.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+} else if (process.env.APP === 'app2') {
+    projectId = 'db096609-2c67-431d-b83d-88cabf8511b1';
```
Comment (brentvatne) [inline eas-config-mismatch.md:17]:
```suggestion
  projectId = 'db096609-2c67-431d-b83d-88cabf8511b1';
```

File: eas-config-mismatch.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    projectId,
+  }
+}
+```
+
+It's not correct to run `APP=app1 eas build --profile production` where **eas.json** is:
+
+```json
+{
+    "build": {
```
Comment (brentvatne) [inline eas-config-mismatch.md:32]: let's use 2 spaces here for indentation for consistency

File: eas-config-mismatch.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        "production-app2": {
+            "env": {
+                "APP": "app2"
+            }
+        }
+    }
+}
+```
+
+> NOTE: Above advice only applies to the build command and other commands that accept the build profile as a parameter, for other commands you will still need to pass environment variable in the cli e.g. `APP=app1 eas update`.
```
Comment (brentvatne) [inline eas-config-mismatch.md:58]:
```suggestion
> NOTE: The above advice only applies to the build command and other commands that accept the build profile as a parameter — for other commands that run entirely locally, you will still need to pass environment variable in the cli e.g. `APP=app1 eas update`.
```

File: eas-config-mismatch.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+
+### Building from GitHub (manual builds triggered from expo.dev website)
+
+When running builds from GitHub there is no local environment that might cause inconstancies, but it's still possible that you might connect a repository to the wrong EAS project or misconfigure connection to multiple projects.
```
Comment (brentvatne) [inline eas-config-mismatch.md:63]:
```suggestion
When running builds from GitHub there is no local environment that might cause inconsistencies, but it's still possible that you might connect a repository to the wrong EAS project or misconfigure a connection to multiple projects.
```

File: eas-config-mismatch.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+If you are connecting single GitHub repository to multiple EAS projects you need to make sure that projectId/slug/owner is matching the correct value for a project that build is executed on.
```
Comment (brentvatne) [inline eas-config-mismatch.md:65]:
```suggestion
If you are connecting single GitHub repository to multiple EAS projects you need to make sure that the `projectId`/`slug`/`owner` all match the correct value for a project that build is executed on.
```

File: eas-config-mismatch.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
+# Evaluating app config on EAS Build workers.
```
Comment (brentvatne) [inline eas-config-mismatch.md:1]:
```suggestion
# Evaluating app config on EAS Build workers
```

**Review by brentvatne — CHANGES_REQUESTED** [review]

**Pull request merged**